core/src/main/java/org/neo4j/graphalgo/core/WeightMap.java

```diff
@@ -2,48 +2,36 @@
 
 import com.carrotsearch.hppc.LongDoubleHashMap;
 import com.carrotsearch.hppc.LongDoubleMap;
+import org.neo4j.graphalgo.api.WeightMapping;
 
 /**
  * single weight cache
  */
-public final class WeightMapping {
+public final class WeightMap implements WeightMapping {
 
     private final int capacity;
     private LongDoubleMap weights;
```

Comment:
for efficiency reasons we might want to look into one that's backed by an int -> int map.
But we can file that as an issue for future development